### DISCUSSION TITLE
defineLegacyParams ()

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#17798] [Q&A General] [answered]
MOOSE group,
I am learning the MOOSE program. Please help me.
I see in the header file:
template <>
InputParameters validParams<DirichletBC>();
Then, in the source file:
defineLegacyParams (DirichletBC);

I don't see them for ADDirichletBC.
Please let me know why defineLegacyParams (ADDirichletBC) is not needed for the AD version.
Thomas

**Comment by GiudGiud** [comment, marked as answer]
Hi Thomas

These is legacy code that is kept only in older object for compatibility with applications that would not have updated the way they use input parameters.

All new objects do not need this, and it's probably why the AD version, which came later, does not have it. 

Guillaume

**Reply by styyokuda** [reply]
Thank you very much Guillaume!
Thomas
